src/lists.rs

```diff
@@ -167,21 +169,35 @@ pub fn definitive_tactic<'t, I, T>(items: I,
                                  .into_iter()
                                  .any(|item| item.as_ref().has_line_pre_comment());
 
-    let limit = match tactic {
+    match tactic {
         _ if pre_line_comments => return DefinitiveListTactic::Vertical,
         ListTactic::Mixed => return DefinitiveListTactic::Mixed,
         ListTactic::Horizontal => return DefinitiveListTactic::Horizontal,
         ListTactic::Vertical => return DefinitiveListTactic::Vertical,
-        ListTactic::LimitedHorizontalVertical(limit) => ::std::cmp::min(width, limit),
-        ListTactic::HorizontalVertical => width,
-    };
+        ListTactic::LimitedHorizontalVertical(..) | ListTactic::HorizontalVertical => {}
+    }
+
+    let item_width_vec = items.clone()
+                              .into_iter()
+                              .map(|item| total_item_width(item.as_ref()))
+                              .collect::<Vec<_>>();
+    let total_width = item_width_vec.iter().fold(0, |acc, x| acc + x);
+    let sep_count = item_width_vec.len();
 
-    let (sep_count, total_width) = calculate_width(items.clone());
     let sep_len = ", ".len(); // FIXME: make more generic?
     let total_sep_len = sep_len * sep_count.checked_sub(1).unwrap_or(0);
     let real_total = total_width + total_sep_len;
 
-    if real_total <= limit && !pre_line_comments &&
+    let horizont_check = match tactic {
+        ListTactic::LimitedHorizontalVertical(limit, l) => {
+            item_width_vec.iter().skip(1).all(|&w| w <= l) ||
+            item_width_vec.iter().rev().skip(1).all(|&w| w <= l) || real_total <= limit
```

Comment:
That takes basically all but the last elements and checks that they are not longer than a given limit.